### ISSUE TITLE
Some questions about the definition of the loss function

### ISSUE THREAD

**Issue by chenZxb** [jiangqy/DPSH-pytorch#9]
The loss function is defined as
       theta_x = train_outputs.mm(Variable(U).t()) / 2
       logloss = (Variable(S)*theta_x - Logtrick(theta_x, use_gpu)).sum() \
                       / (num_train * len(train_label))
       regterm = (Bbatch-train_outputs).pow(2).sum() / (num_train * len(train_label))
       loss =  - logloss + lamda * regterm
Why don't you use Variable(U).mm(Variable(U).t()) / 2 (All train data)  or 
train_outputs.mm(train_outputs.t()) / 2 (The mini-batch data ) when you calculate theta_x? In this condition，Sij can be the pairwise label between all train data or the mini-batch data. May I ask if there is any reason why you have to do so? I'm trying to rewrite the loss function in my way,but I get a huge loss. It even prints NaN,I don't know why this is happening.

**Comment by jiangqy** [comment]
These two sampling strategies are okay. We adopt column sampling in this paper. Hence I follow the setting of the paper.
